tests/Unit/Traits/HasPreAndPostProcessingTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Native\Electron\Tests\Unit\Traits;
+
+use Illuminate\Support\Facades\Config;
+use Native\Electron\Tests\Mocks\HasPreAndPostProcessingMock;
+
+it('can run pre and post processing from config', function () {
+    $tmpDir = sys_get_temp_dir();
+
+    Config::set('nativephp.prebuild', [
+        'touch '.$tmpDir.'/prebuild1',
+        'touch '.$tmpDir.'/prebuild2',
+    ]);
+
+    Config::set('nativephp.postbuild', [
+        'touch '.$tmpDir.'/postbuild1',
+        'touch '.$tmpDir.'/postbuild2',
+    ]);
+
+    $mock = \Mockery::mock(HasPreAndPostProcessingMock::class)
+        ->makePartial();
+
+    $mock->shouldAllowMockingProtectedMethods();
+    $mock->shouldReceive('getPrePostBuildConfig')
+        ->with('prebuild')
+        ->andReturn(collect(Config::get('nativephp.prebuild')));
+
+    $mock->shouldReceive('getPrePostBuildConfig')
+        ->with('postbuild')
+        ->andReturn(collect(Config::get('nativephp.postbuild')));
```

Comment:
Great that this is tested! I see you are actually running the commands to check if files were created. That is great!

There is a preexisting pattern we already use you can do this with which doesn't use mockery (and is a bit more expressive)

This uses a anonymous class that uses the trait you can just call from your test. Since the trait is isolated as is I don't think we need to actually mock any methods.

Bonus: we can still introduce mockery later if we need it